### PR TITLE
feat: documentation versioning implementation

### PR DESCRIPTION
**Pull request by iamnamananand996**
Because

- documentation versioning implementation

This commit

- documentation versioning implementation

### CODE REVIEW

**Review by EiffelFly — COMMENTED** [review]
There needs some small changes, please check my comment

**Comment by iamnamananand996** [comment]
Hi @EiffelFly , update suggested changes, will make some more changes for the version dropdown.

**Comment by EiffelFly** [comment]
@iamnamananand996 From the UX point of view. We should not have a version folder name called latest. Because

- Whenever the user has a new version he will need to change the folder name and rename the correct version to the latest. Then an issue will arise, what is the exact version name of this latest folder

The ideal workflow will be

- User adds the new version with the correct version name
- User indicates the latest alias on that version (They point v1.0.0 to the latest in our content.config.ts)
- We auto replace the v1.0.0 with the latest tag

**Comment by iamnamananand996** [comment]
okay, I will make this change

**Comment by iamnamananand996** [comment]
Hi @EiffelFly, update the suggested changes

File: content.config.tsx [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
 import {
   DiscordIcon,
   GitHubIcon,
   Logos,
   Icons,
+  Tag,
 } from "@instill-ai/design-system";
 
-import { useTranslation } from "next-i18next";
+export const VERSIONS: any = {
```
Comment (EiffelFly) [inline content.config.tsx:13]: @iamnamananand996 I think we can use this type here

```
Record<string, string>
```

Comment (EiffelFly) [reply content.config.tsx:13]: @iamnamananand996 On a second thought I think we need to make it as explicit as possible

const LATEST_VERSION = "v1.0.0"

File: next.config.mjs [hunk 2]
```diff
@@ -16,6 +16,10 @@ import {
 } from "./src/lib/markdown/rehype-info-block-handler.mjs";
 import path from "path";
 
+const VERSIONS = {
+  latest: "1.0.0",
+};
+
```
Comment (EiffelFly) [inline next.config.mjs:22]: @iamnamananand996 What is the purpose of the duplicated declaration? (content.config.ts already had one)

File: src/lib/instill/applicationVersion.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { InstillAppVersion, Nullable } from "@/types/instill";
+import { VERSIONS } from "../../../content.config";
+
+export function getApplicationVersion(
+  path: string | Array<string>
+): InstillAppVersion {
+  if (path.includes("1.0.0")) {
```
Comment (EiffelFly) [inline src/lib/instill/applicationVersion.ts:7]: @iamnamananand996 This should not be hard-coded

**Review by EiffelFly — COMMENTED** [review]
@iamnamananand996 I think the logic of previous solution about the used-defined latest version is a bit messy. Please clean them up

- User defined what the latest version is
- The docs folder should not have any folder with latest folderVersionName as /docs/base/lates
- The latest is an alias only for display purpose

**Comment by iamnamananand996** [comment]
Hi @EiffelFly , I have updated the suggested changes.

**Comment by iamnamananand996** [comment]
Hi @EiffelFly, update the changes now

File: version.mjs
Comment (EiffelFly) [inline version.mjs:1]: @iamnamananand996 We should put this configuration together with content.config.ts

File: src/middleware.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    const redirectUrl = request.url.replace("latest", VERSIONS["cloud"]);
+    return NextResponse.redirect(new URL(redirectUrl, request.url));
+  }
+  if (request.nextUrl.pathname.startsWith("/docs/model/latest")) {
+    const redirectUrl = request.url.replace("latest", VERSIONS["model"]);
+    return NextResponse.redirect(new URL(redirectUrl, request.url));
+  }
+}
+
+// See "Matching Paths" below to learn more
```
Comment (EiffelFly) [inline src/middleware.ts:29]: @iamnamananand996 We can remove this comment

**Review by EiffelFly — COMMENTED** [review]
@iamnamananand996 All he dropdown should close once the action is finished. Other than that I added some minor suggestion, please take a look. let aim to merge this PR by today 

https://github.com/instill-ai/instill.tech/assets/57251712/f38cca35-a23d-49d2-9a0c-a6b6608d4a40

File: version.mjs
Comment (iamnamananand996) [reply version.mjs:1]: yes, you are correct, but `next.config.mjs` uses these version for redirect links,
and their `.ts` import does not compile properly, so have to add it like this.

**Review by EiffelFly — APPROVED** [review]
LGTM

**Pull request merged**